pkg/apis/submariner.io/v1/endpoint.go

```diff
@@ -120,6 +120,9 @@ func getIPFrom(family k8snet.IPFamily, ips []string, ipv4Fallback string) string
 
 func setIP(ips []string, ipv4Fallback, newIP string) ([]string, string) {
 	family := k8snet.IPFamilyOfString(newIP)
+	if family == k8snet.IPFamilyUnknown {
```

Comment:
Wouldn’t it be better to return an error?

Reply:
Perhaps for an invalid IP but then every caller has to check the error even though it's likely not possible for it to pass an invalid IP. Handling an empty IP without failing avoids more generic client code from having to first verify non-empty.

Reply:
OK, so if this is about handling empty strings, I think it’s best to be explicit about it:

```
func setIP(ips []string, ipv4Fallback, newIP string) ([]string, string) {
	if newIP == "" {
		return ips, ipv4Fallback
	}

	family := k8snet.IPFamilyOfString(newIP)
	…
```

Handling `IPFamilyUnknown` suggests that it’s about handling invalid addresses, not IPs (at least, to me).

Reply:
Or add a comment mentioning that `IPFamilyUnknown` corresponds to empty addresses :wink:.

Reply:
Checking `IPFamilyUnknown` handles both empty and invalid as neither translates to V4 or V6. I'll add explicit check for empty so that `IPFamilyUnknown` means invalid and log an error message.